stubs/pygit2/pygit2/callbacks.pyi

```diff
@@ -0,0 +1,79 @@
+from _typeshed import AnyStr_co
+from collections.abc import Callable
+from contextlib import AbstractContextManager
+from os import PathLike
+from typing import Protocol
+from typing_extensions import ParamSpec, Self, TypeAlias
+
+from _cffi_backend import _CDataBase
+
+from ._pygit2 import DiffFile, Oid
+from .enums import CheckoutNotify, CheckoutStrategy, CredentialType, StashApplyProgress
+from .remotes import TransferProgress
+from .utils import _IntoStrArray
+
+class Payload:
+    def __init__(self, **kw: dict[str, object]) -> None: ...
```

Comment:
```suggestion
    def __init__(self, **kw: object) -> None: ...
```
Note that the type for `**kwargs` is the type of individual kwargs, not of the whole kwargs dict.